streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1267,7 +1267,7 @@ public RecordQueue createQueue(final TopicPartition partition) {
             final SourceNode<?, ?> source = topology.source(partition.topic());
             if (source == null) {
                 throw new TopologyException(
-                        "Topic is unknown to the topology. " +
+                        "Topic " + partition.topic() + " is unknown to the topology. " +
```

Comment:
This is piggy-backed improvement when I tried to debug the failure: it's better to log the missing topic name as well.